### PR TITLE
fIX: if any empty charater around comma

### PR DESCRIPTION
**Pull request by DevXiaolan**
see :  [test/empty_around_comma.js](https://github.com/DevXiaolan/cron-parser/blob/master/test/empty_around_comma.js)

### CODE REVIEW

**Comment by DevXiaolan** [comment]
Hi ?

**Comment by harrisiirak** [comment]
@DevXiaolan thanks for your contribution and pointing out this issue!

Ideally, as this pattern `* * * * ,` (or `* * * * ,2`) is actually invalid, we should be throwing an error instead. Converting invalid statement into wildcard (*) may be a little bit confusing, therefore throwing a validation error may be more meaningful here.

**Comment by DevXiaolan** [comment]
> @DevXiaolan thanks for your contribution and pointing out this issue!
> 
> Ideally, as this pattern `* * * * ,` (or `* * * * ,2`) is actually invalid, we should be throwing an error instead. Converting invalid statement into wildcard (*) may be a little bit confusing, therefore throwing a validation error may be more meaningful here.

Maybe you are right, I'll follow your advice later

**Comment by DevXiaolan** [comment]
keep waiting

**Comment by harrisiirak** [comment]
@DevXiaolan thanks for the update! I'll give a look.

Please use "Request Review" then I actually know when things are ready to review:
Screenshot 2020-11-16 at 09 10 01

File: lib/expression.js [hunk 1]
```diff
@@ -248,7 +253,11 @@ CronExpression._parseField = function _parseField (field, value, constraints) {
     var atoms = val.split(',');
     if (atoms.length > 1) {
```
Comment (harrisiirak) [inline lib/expression.js:254]: Actually, I was playing around with this validation and I found that we can use a little bit simpler approach here. Let me know what you're thinking about this.

```suggestion
    if (!atoms.every(function (atom) {
      return atom.length > 0;
    })) {
      throw new Error('Invalid list value format');
    }

    if (atoms.length > 1) {
```

File: lib/expression.js [hunk 2]
```diff
@@ -248,7 +253,11 @@ CronExpression._parseField = function _parseField (field, value, constraints) {
...
       for (var i = 0, c = atoms.length; i < c; i++) {
-        handleResult(parseRepeat(atoms[i]));
+        if(atoms[i]){
```
Comment (harrisiirak) [inline lib/expression.js:256]: If this if/else can be remove if my suggestion makes sense.

File: lib/expression.js [hunk 3]
```diff
@@ -186,6 +186,11 @@ CronExpression._parseField = function _parseField (field, value, constraints) {
     throw new Error('Invalid characters, got value: ' + value)
   }
 
+  // check if empty around comma
+  if (value === ',') {
+    throw new Error('Invalid cron expression, value can not specified as comma');
```
Comment (harrisiirak) [inline lib/expression.js:191]: The same about this code path if my suggestions seems to fit.

**Review by harrisiirak — CHANGES_REQUESTED** [review]
Found a better way how we can actually simplify this. Let me know what do you think about this.

File: lib/expression.js (hunk 1 above)
Comment (DevXiaolan) [reply lib/expression.js:260]: LGTM

File: lib/expression.js (hunk 3 above)
Comment (DevXiaolan) [reply lib/expression.js:191]: this line pre-check the input values before calling  parseSequence()

Comment (harrisiirak) [reply lib/expression.js:191]: No need for that anymore as validation happens with atoms.every.

File: lib/expression.js [hunk 4]
```diff
@@ -186,6 +186,11 @@ CronExpression._parseField = function _parseField (field, value, constraints) {
...
+  }
+
```
Comment (harrisiirak) [inline lib/expression.js:193]: Still not needed as this special error doesn't create much additional value. If we remove this, it still throws `Invalid list value format` as an error, therefore we have pretty simple validation model to catch both cases.

**Review by harrisiirak — CHANGES_REQUESTED** [review]

File: lib/expression.js (hunk 4 above)
Comment (DevXiaolan) [reply lib/expression.js:193]: 你说的也有道理，

**Review by harrisiirak — APPROVED** [review]
LGTM! Thanks! I'll merge it after Travis run completes.

**Pull request merged**